ext/opentelemetry-ext-django/src/opentelemetry/ext/django/__init__.py

```diff
@@ -0,0 +1,95 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+from os import environ
+
+from django import VERSION
+from django.conf import settings
+
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.configuration import Configuration
+from opentelemetry.ext.django.middleware import OpenTelemetryMiddleware
+
+_logger = getLogger(__name__)
+
+
+class DjangoInstrumentor(BaseInstrumentor):
+    """A instrumentor for flask.Django
+
+    See `BaseInstrumentor`
+    """
+
+    _opentelemetry_middleware = ".".join(
+        [
+            OpenTelemetryMiddleware.__module__,
+            OpenTelemetryMiddleware.__qualname__,
+        ]
+    )
+
+    def __init__(self):
+        super().__init__()
+        # Django Middleware is code that is executed before and/or after a
+        # request. Read about Django middleware here:
+        # https://docs.djangoproject.com/en/3.0/topics/http/middleware/
+
+        # This method must set this Django settings attribute:
+        # MIDDLEWARE: for Django version > 1.0
+        # MIDDLEWARE_CLASSES: for Django version <= 1.0
+
+        # Django settings.MIDDLEWARE is a list of strings, each one a Python
+        # path to a class or a function that acts as middleware.
+
+        self._middleware_setting = (
+            "MIDDLEWARE" if VERSION >= (1, 10, 0) else "MIDDLEWARE_CLASSES"
+        )
+
+    def _instrument(self):
+
+        # FIXME this is probably a pattern that will show up in the rest of the
+        # ext. Find a better way of implementing this.
+        # FIXME Probably the evaluation of strings into boolean values can be
+        # built inside the Configuration class itself with the magic method
+        # __bool__
+        if (
+            hasattr(Configuration(), "django_instrument")
+            and Configuration().django_instrument != "True"
+        ):
+            return
+
+        if "DJANGO_SETTINGS_MODULE" not in environ:
+            raise Exception(
+                "Missing environment variable DJANGO_SETTINGS_MODULE"
+            )
+
+        settings_middleware = getattr(settings, self._middleware_setting, [])
+        settings_middleware.append(self._opentelemetry_middleware)
```

Comment:
This can not be solved, but is an inherent problem of this approach: the order of middleware entries matters, and here you have no control on that. See https://docs.djangoproject.com/en/3.0/topics/http/middleware/#activating-middleware and https://docs.djangoproject.com/en/3.0/ref/middleware/#middleware-ordering

Reply:
Thanks for pointing that out, added that comment in the code itself.